### PR TITLE
Probes with point interpolator

### PR DESCRIPTION
**Pull request by vbaconnet**
Got rid of `findpts_eval` for the interpolation in `probes%interpolate`. Also removed `coef_t` from point_interpolator as it is not using it, and adjusted the RBC example.

### CODE REVIEW

**Review by njansson — APPROVED** [review]

**Comment by timofeymukha** [comment]
Nice, gslib now only searches for elements and then we use the interpolator you added before? For the space variable, I guess it is  canonical to use `Xh`, with a capital x, as an exception to the general rule. This mimics the math notation in some books.

File: src/common/probes.F90 [hunk 1]
```diff
@@ -362,17 +370,18 @@ subroutine probes_interpolate(this, t, tstep, write_output)
...
-                     this%sampled_fields%fields(il)%f%x(:,:,:,this%el_owner(i)))
+                     this%sampled_fields%fields(il)%f%x(:,:,:,this%el_owner(i)+1))
 
                 local_out_field(i) = my_interpolated_field(1)
              end if
           end do
 
           ! Artificial way to gather all values to rank 0
-          call MPI_Reduce(local_out_field, this%out_fields(il,1), this%n_probes, &
+          call MPI_Reduce(local_out_field, this%out_fields(1, il), this%n_probes, &
```
Comment (adperezm) [inline src/common/probes.F90:383]: Smart :)

Comment (vbaconnet) [reply src/common/probes.F90:383]: Yes, and also about out_fields I reshaped the array so we don't need to transpose it for the csv_file output

**Review by MartinKarp — APPROVED** [review]
Tremendous work Bacon 1!

**Pull request merged**